src/spring/azext_spring/custom.py

```diff
@@ -1462,3 +1467,49 @@ def app_insights_show(cmd, client, resource_group, name, no_wait=False):
     if not monitoring_setting_properties:
         raise CLIError("Application Insights not set.")
     return monitoring_setting_properties
+
+
+def app_connect(cmd, client, resource_group, service, name,
+                deployment=None, instance=None, shell_cmd='/bin/sh'):
+
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    creds, _, _ = profile.get_raw_token()
+    token = creds[1]
+
+    resource = client.services.get(resource_group, service)
+    hostname = resource.properties.fqdn
+    if not instance:
+        if not deployment.properties.instances:
+            raise CLIError("No instances found for deployment '{0}' in app '{1}'".format(
+                deployment.name, name))
```

Comment:
Could you please use the specific error type from `azure.cli.core.azclierror` instead of `CLIError`?

Reply:
Updated.